### PR TITLE
Api/missions api

### LINKED ISSUE #56: [4pt]  Fetch missions - Fetch data
Fetch data from the Missions endpoint  (https://api.spacexdata.com/v3/missions) when a user navigates to the Missions section.

Once the data are fetched, dispatch an action to store the selected data in Redux store:

- mission_id
- mission_name
- description

NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### PR DESCRIPTION
**Pull request by Mo7amedSMahdi**
- Add redux middleware.
- use the middleware to fetch the data from API.
- use loader for lazy loading or slow internet connection.
- Prevent from fetching the data if the store contains data or the last fetch is less than 10 minutes.

### CODE REVIEW

File: src/store/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
+import { combineReducers } from 'redux';
+// import thunk from 'redux-thunk';
```
Comment (ismailakinkunmi) [inline src/store/configureStore.js]: please do not drop unused code as a comment in your source code, unused code makes code not clean.

File: src/store/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import api from './middleware/api';
+import missionsReducer from './missions';
+
+const rootReducer = combineReducers({ missions: missionsReducer });
+
+const store = configureStore({
+  reducer: rootReducer,
+  middleware: [
+    ...getDefaultMiddleware(),
+    // logger({ destination: "console" }),\
```
Comment (ismailakinkunmi) [inline src/store/configureStore.js]: please do not drop unused code as a comment in your source code, as it make code not clean

**Review by ismailakinkunmi — COMMENTED** [review]
Congrat, I love your design  and  you're  a great designer
please don't forget to remove those unnecessary comments before you merge.

**Review by ismailakinkunmi — APPROVED** [review]

File: src/store/configureStore.js (hunk 1 above)
Comment (Mo7amedSMahdi) [reply src/store/configureStore.js]: Done 👍

File: src/store/configureStore.js (hunk 2 above)
Comment (Mo7amedSMahdi) [reply src/store/configureStore.js]: Done 👍

**Pull request merged**